packages/toolpad-utils/src/collections.ts

```diff
@@ -93,3 +114,65 @@ export function filterKeys<U>(
 ): Record<string, U> {
   return mapProperties(obj, ([key, value]) => (filter(key) ? [key, value] : null));
 }
+
+/**
+ * check if two objects are deeply equal and support  Set, Map, Date, Array, Object
+ */
+// Reference code https://github.com/vuejs/vue/blob/49b6bd4264c25ea41408f066a1835f38bf6fe9f1/src/shared/util.ts#L297
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function isDeepEqual(obj: any, objToCompare: any): boolean {
```

Comment:
So, the idea behind this ticket was not to one to one recreate lodash. It's about evaluating why the algorithm needs a nested equality check, and removing that need. The fact that we use this function seems to be rather a symptom that we're using a less than ideal algorithm.

I mean, if vendoring in some lodash functions would be all that's needed for this ticket, we would have simply done it and not opened a ticket at all.

Reply:
> It's about evaluating why the algorithm needs a nested equality check, and removing that need.

I apologized. I do not totally understand this sentence.

Reply:
If I get it right, you want to delete some redundant code that is not related to our industry, don’t you? Such as checking the binary and string type.

Reply:
We don't want to use lodash all over our codebase because
1. 90% of it can already be achieved using standard javsacript. No need for `_.map`, `_.filter`, ... it all exists in javascript. no need for `_.isEmpty`, `_.isObject`,... because you can write all of these with simple one-liners that everybody knows and understands. 
2. it is unmaintained.
3. its api design makes programmers turn type-safe code into type-unsafe code.

For this same reason we are not going to create a library like lodash, we will just simply aim at removing lodash.